### PR TITLE
Extract the game rules into a separate file.

### PR DESCRIPTION
**Pull request by k-owens**
I am doing some major refactoring.  I am putting the refactored code into different files than the 3X3 and 4X4.  I am keeping the 3X3 and the 4X4 until I have taken everything out of them, and then I will delete them.

### CODE REVIEW

**Comment by aaronlahey** [comment]
Read this: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Comment by aaronlahey** [comment]
Please put the description you posted in slack in either the commit message extended description (after the first line) or in the PR description.

**Comment by aaronlahey** [comment]
Do things included in `.vs`, `bin`, or `obj` directories need to be under source control? This is pretty large number of files changed.

File: TicTacToe.test/UnitTests.fs [hunk 1]
```diff
@@ -3,481 +3,60 @@
 open Xunit
 open GamePlayer3X3
 open GamePlayer4X4
+open GameBoard
 
 [<Fact>]
-let canMakeMoveSpot1 () =
```
Comment (aaronlahey) [inline TicTacToe.test/UnitTests.fs:8]: I see a lot of tests deleted. Where did those go?

**Comment by aaronlahey** [comment]
LGTM with feedback addressed.

**Comment by k-owens** [comment]
I will add a .gitignore to deal with all of the extra files. In regards to the tests, a lot of them seem unnecessary now.  There were tests for both 3X3 and 4X4.  The code will now be able to handle any type of board.  I took the rest out and I am slowly adding them back in as I refactor the code.

**Pull request merged**